lib/internal/inspector_async_hook.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+const { createHook } = require('async_hooks');
+const inspector = process.binding('inspector');
+const errors = require('internal/errors');
+
+if (!inspector || !inspector.asyncTaskScheduled) return;
+
+module.exports = createHook({
+  init(asyncId, type, triggerId, resource) {
+    if (!Number.isInteger(asyncId))
+      throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'asyncId', 'integer');
+    if (typeof type !== 'string')
+      throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'type', 'string');
+    // TODO(bajtos): support recurring tasks like setInterval
```

Comment:
I am not aware of any better documentation than the source code. I don't have any strong opinion on what's the best value to use, I am ok with using `true`.